packages/cyberstorm/src/components/Header/UserDropDown.tsx

```diff
@@ -17,27 +11,17 @@ import { AvatarButton } from "../Avatar/AvatarButton";
 import * as Button from "../Button/";
 import { DropDown, DropDownDivider, DropDownItem } from "../DropDown/DropDown";
 import { DropDownLink } from "../DropDown/DropDownLink";
-import { SettingsLink, TeamsLink, UserLink } from "../Links/Links";
 
-/**
- * TODO: show fallback if user is not authenticated.
- * TODO: check if all the features in dropdown are actually supported
- *       on beta launch.
- */
 export const UserDropDown = () => {
   const dapper = useDapper();
   const user = usePromise(dapper.getCurrentUser, []);
   const avatar = user.connections.find((c) => c.avatar !== null)?.avatar;
 
-  if (!user.username) {
```

Comment:
Irrelevant now, but as more stuff gets added the if-else approach below gets quite messy. Would probably be cleaner to have two subcomponents that both return a DropDown and then just to something like `return user.username ? <AuthenticatedDropDown user={user} /> : <UnauthenticatedUser />`

Reply:
Yea, I've tried to improve this component structure in a later PR, but this definitely needs more work in the future. Someone might call this code debt for when the other pages are added, but I call it iterative development 👓